source/engine/graphics/api/vulkan/RenderGraph.cpp

```diff
@@ -50,12 +54,44 @@ void SetViewportAndScissorAspectRatio(vk::CommandBuffer* cmd, const nc::Vector2&
     cmd->setScissor(0, 1, &scissor);
 }
 
-auto CreateLitPass(const nc::graphics::Device& device, nc::graphics::GpuAllocator *allocator, nc::graphics::Swapchain *swapchain, const nc::Vector2& dimensions) -> nc::graphics::RenderPass
+auto CreateShadowMappingPass(const nc::graphics::Device* device, nc::graphics::GpuAllocator* allocator, nc::graphics::Swapchain* swapchain, const nc::Vector2& dimensions, uint32_t index) -> nc::graphics::RenderPass
 {
     using namespace nc::graphics;
 
-    const auto vkDevice = device.VkDevice();
-    const auto& gpuOptions = device.GetGpuOptions();
+    const auto vkDevice = device->VkDevice();
+
+    const auto shadowAttachmentSlots = std::array<AttachmentSlot, 1>
```

Comment:
Looks like this could be `constexpr`.

Reply:
I get a "doesn't evaluate to a constant expression" for all of these